modules/terraform/options.go

```diff
@@ -37,3 +40,54 @@ type Options struct {
 	Logger                   *logger.Logger         // Set a non-default logger that should be used. See the logger package for more info.
 	Parallelism              int                    // Set the parallelism setting for Terraform
 }
+
+// Clone makes a deep copy of the Options object and returns it.
+func (options *Options) Clone() (*Options, error) {
+	newOptions := Options{}
+	if err := copier.Copy(&newOptions, options); err != nil {
+		return nil, err
+	}
+	return &newOptions, nil
+}
+
+// WithDefaultRetryableErrors makes a copy of the Options object and returns an updated object with sensible defaults
+// for retryable errors. The included retryable errors are typical errors that most terraform modules encounter during
+// testing, and are known to self resolve upon retrying.
+// This will fail the test if there are any errors in the cloning process.
+func WithDefaultRetryableErrors(t *testing.T, originalOptions *Options) *Options {
+	newOptions, err := originalOptions.Clone()
+	require.NoError(t, err)
+
+	retryableTerraformErrors := map[string]string{
```

Comment:
NIT: Should this be extracted into a variable that's shared, in case someone wants to use it outside of this method?

Reply:
Good idea. Done! e5c4287